onnx/defs/math/defs.cc

```diff
@@ -416,6 +471,117 @@ will throw errors.
+  .TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain input and output types to float tensors.");
+
+OPERATOR_SCHEMA(Hardmax)
+  .NumInputs(1)
+  .NumOutputs(1)
+  .SetDoc(R"DOC(
+The operator computes the hardmax values for each layer in the batch
+ of the given input. The input is a 2-D tensor (Tensor<float>) of size
+(batch_size x input_feature_dimensions). The output tensor has the same shape
+and contains the softmax normalized values of the corresponding input.
+
+X does not need to explicitly be a 2D vector; rather, it will be
+coerced into one. For an arbitrary n-dimensional tensor
+X \in [a_0, a_1, ..., a_{k-1}, a_k, ..., a_{n-1}] and k is
+the axis provided, then X will be coerced into a 2-dimensional tensor with
+dimensions [a_0 * ... * a_{k-1}, a_k * ... * a_{n-1}]. For the default
+case where axis=1, this means the X tensor will be coerced into a 2D tensor
+of dimensions [a_0, a_1 * ... * a_{n-1}], where a_0 is often the batch size.
+In this situation, we must have a_0 = N and a_1 * ... * a_{n-1} = D.
+Each of these dimensions must be matched correctly, or else the operator
+will throw errors.
+)DOC")
+  .Attr("axis",
+        "(int) default to 1; describes the axis of the inputs when coerced "
+        "to 2D; defaults to one because the 0th axis most likely describes "
+        "the batch_size",
+        AttrType::INT)
+  .Input(0, "input",
+         "The input tensor that's coerced into a 2D matrix of size (NxD) "
+         "as described above.", "T")
+  .Output(0, "output", "The hardmax normalized output values with the same "
+          "shape as input tensor.", "T")
+  .TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain input and output types to float tensors.");
+
+OPERATOR_SCHEMA(Softsign)
+    .NumInputs(1)
+    .NumOutputs(1)
+    .SetDoc(R"DOC(
+Calculates the softsign (x/1+|x|) of the given input tensor element-wise. This
+operation can be done in an in-place fashion too, by providing the same input
+and output blobs.
+)DOC")
+    .Input(0, "input", "1-D input tensor", "T")
+    .Output(
+        0,
+        "output",
+        "The softsign (x/1+|x|) values of the input tensor computed element-wise",
+        "T")
+    .TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain input and output types to float tensors.");
+
+OPERATOR_SCHEMA(Softplus)
+    .NumInputs(1)
+    .NumOutputs(1)
+    .SetDoc(R"DOC(
+Softplus takes one input data (Tensor<T>) and produces one output data
+(Tensor<T>) where the softplus function, y = ln(exp(x) + 1), is applied to
+the tensor elementwise.
+)DOC")
+    .Input(0, "X", "1D input tensor", "T")
+    .Output(0, "Y", "1D input tensor", "T")
+    .TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain input and output types to float tensors.");
+
+OPERATOR_SCHEMA(ParametricSoftplus)
```

Comment:
can we merge ParametricSoftplus and Softplus? They are less frequently used so fewer ops - the better